master/docs/manual/cfg-reporters.rst

```diff
@@ -109,137 +109,38 @@ If you want to require Transport Layer Security (TLS), then you can also set ``u
-            rev_steps = reversed(build.getSteps())
-            # find the last step that finished
-            for step in rev_steps:
-                if step.isFinished():
-                    break
-            # get logs for the last finished step
-            if step.isFinished():
-                logs = step.getLogs()
-            # No step finished, loop just exhausted itself; so as a special case we fetch all logs
-            else:
-                logs = build.getLogs()
-            # logs within a step are in reverse order. Search back until we find stdio
-            for log in reversed(logs):
-                if log.getName() == 'stdio':
-                    break
-            name = "%s.%s" % (log.getStep().getName(), log.getName())
-            status, dummy = log.getStep().getResults()
-            # XXX logs no longer have getText methods!!
-            content = log.getText().splitlines() # Note: can be VERY LARGE
-            url = u'%s/steps/%s/logs/%s' % (master_status.getURLForThing(build),
-                                           log.getStep().getName(),
-                                           log.getName())
-
-            text.append(u'<i>Detailed log of last build step:</i> <a href="%s">%s</a>'
-                        % (url, url))
-            text.append(u'<br>')
-            text.append(u'<h4>Last %d lines of "%s"</h4>' % (limit_lines, name))
-            unilist = list()
-            for line in content[len(content)-limit_lines:]:
-                unilist.append(cgi.escape(text_type(line,'utf-8')))
-            text.append(u'<pre>')
-            text.extend(unilist)
-            text.append(u'</pre>')
-            text.append(u'<br><br>')
-            text.append(u'<b>-The Buildbot</b>')
-            return {
-                'body': u"\n".join(text),
-                'type': 'html'
-            }
+                                messageFormatter=reporters.MessageFormatter(template="STATUS: {{ summary }}"))
+
+Another example of a function delivering a customized html email is given below::
+
+    from buildbot.plugins import reporters
+
+    # XXX: This is work in progress, use with care.
+    template=u'''\
+    <h4>Build status: {{ summary }}</h4>
+    <p> Worker used: {{ workername }}</p>
+    {% for step in build['steps'] %}
+    <p> {{ step['name'] }}: {{ step['result'] }}</p>
+    {% endfor %}
+    <b> -- The Buildbot</p>
+    '''
 
     mn = reporters.MailNotifier(fromaddr="buildbot@example.org",
                                 sendToInterestedUsers=False,
                                 mode=('failing',),
                                 extraRecipients=['listaddr@example.org'],
-                                messageFormatter=html_message_formatter)
+                                messageFormatter=reporters.MessageFormatter(
+                                    template=template, template_type='html',
+                                    wantProperties=True, wantSteps=True, wantLogs=True))
```

Comment:
I think while we are at it we should really document that MessageFormatter, describing all the available parameters and their implication.
The more details you ask via wantXXX the more overhead the mail reporter will have (memory and CPU)
There is a completely obsolete documentation of messageFormater later in this file